packages/validation/src/validators/custom/simpleValidator.ts

```diff
@@ -37,10 +46,14 @@ export class SimpleCustomValidator extends AbstractPlugin {
     incremental?: Incremental
   ): Promise<ValidationResult[]> {
     const results: ValidationResult[] = [];
+    const resourceMap = keyBy(resources, (r) => r.id);
 
+    const clonedResources: PlainResourceWithId[] = resources.map((r) =>
```

Comment:
I worry a bit about the performance of this but it's dangerous as all these resources end up straight in our Redux again since we pass by reference.. Let's see what this gives, I think later I'll want to make a single copy at the start of the validation.